source/models/Loadable/index.js

```diff
@@ -0,0 +1,72 @@
+// eslint-disable-next-line no-restricted-imports
+import Model from '../Model';
+
+export const LOADABLE_STATES = Object.freeze({
+    UNINITIALIZED: 'UNINITIALIZED',
+    IN_PROGRESS: 'IN_PROGRESS',
+    COMPLETED: 'COMPLETED',
+    ERROR: 'ERROR',
+});
+const LOADABLE_STATES_VALUES = Object.values(LOADABLE_STATES);
+
+const LOADABLE_MODEL_KEYS = Object.freeze({
+    state: '_loadable_state',
+});
+
+const loadableSchema = {
+    [LOADABLE_MODEL_KEYS.state]: LOADABLE_STATES.UNINITIALIZED,
+};
+
+const LoadableModel = (defaultValues, name) => class extends Model({
+    ...defaultValues,
+    ...loadableSchema,
+}, name) {
+    _getLoadingState() {
+        return this.get(LOADABLE_MODEL_KEYS.state);
+    }
+
+    _withLoadingState(state) {
+        if (LOADABLE_STATES_VALUES.includes(state)) {
+            return this.set(LOADABLE_MODEL_KEYS.state, state);
+        }
+
```

Comment:
That's up to your thunk/action - this is only a collection of methods that are easing handling of the loading state.

Reply:
This being treated as third party code seems like even more of a reason to add an error message, since it catches if a developer is misusing the library. As it stands, this fails silently.